ipatests/test_integration/test_commands.py

```diff
@@ -1622,6 +1704,74 @@ def test_ipa_systemd_journal(self):
         )
         self.master.run_command(restore_cmd)
 
+    def test_ipa_cacert_manage_duplicate_certsubject(self):
+        """Test for ipa-cacert-manage install with duplicated
+           certificate subjects. This relies on the behavior
+           of NSS to show the certificates separately rather than
+           lumping the duplicates together. This requires different
+           validity periods, say 3 years + 1 day.
+        """
+
+        certfile = os.path.join(self.master.config.test_dir, 'chain.pem')
+        self.master.put_file_contents(certfile, originalsubjectchain)
+        result = self.master.run_command(
+            [paths.IPA_CACERT_MANAGE, 'install', certfile])
+
+        certs = self.master.run_command(
+            [paths.IPA_CACERT_MANAGE, 'list'], raiseonerr=False
+        ).stdout_text
+
+        assert f"{interm_nick}  {intermediate_serial}" in certs
+
+        certfile = os.path.join(self.master.config.test_dir, 'interm.pem')
+        self.master.put_file_contents(certfile, duplicatesubject)
+        result = self.master.run_command(
+            [paths.IPA_CACERT_MANAGE, 'install', certfile])
+
+        certs = self.master.run_command(
+            [paths.IPA_CACERT_MANAGE, 'list'], raiseonerr=False
+        ).stdout_text
+
+        # If the duplicate subject certificates are not sufficiently
+        # different in validity period, or prior to the this fix,
+        # the test will fail because only one of the duplicately named
+        # subject certificates will be visible: the second one (4097).
+        assert f"{interm_nick}  {intermediate_serial}" in certs
+        assert f"{interm_nick}  {duplicate_serial}" in certs
+
+        # Make sure we can install the new certs systemwide
+        # No assertions needed, it will work or it won't
+        self.master.run_command(["ipa-certupdate"])
+
+        # delete one of the duplicate subjects, no serial number
+        result = self.master.run_command(
+            ['ipa-cacert-manage', 'delete', interm_nick],
+            raiseonerr=False
+        )
+        assert result.returncode == 1
+        assert 'Multiple matching certificates' in result.stderr_text
+
+        # delete one of the duplicate subjects by the serial number
+        result = self.master.run_command(
+            ['ipa-cacert-manage', 'delete', interm_nick,
+             '--serial', intermediate_serial,],
+            raiseonerr=False
+        )
+        assert result.returncode == 0
+
+        certs = self.master.run_command(
+            [paths.IPA_CACERT_MANAGE, 'list'], raiseonerr=False
+        ).stdout_text
+
+        assert f"{interm_nick}  {intermediate_serial}" not in certs
+        assert f"{interm_nick}  {duplicate_serial}" in certs
+
+    def test_ipa_cacert_manage_prune(self):
```

Comment:
The rebase to master branch introduced an error. The test test_ipa_cacert_manage prune has been moved to TestIPACommandWithoutReplica and should not be here.

Your new test has been added in the middle of test_ipa_systemd_journal.

Reply:
Yeesh, I really botched that merge royally. I fixed this by running diffs between a saved copy of the file from master vs the abomination I created and fixed things up from there.